CONTRIBUTING.md

```diff
@@ -1,174 +1,163 @@
-to a code review.
-	- Nobody writes perfect code first time - we're all humans. You will almost
-	certainly be asked to make changes as part of this process, so please be
-	patient with us, and make sure to voice your opinion if you disagree with
-	any suggested changes or rationale so we can make your PR the best it can be!
-- Alternatively, if we decide your pull request doesn't quite align with Fusion,
-then we'll explain in as much detail as possible why we don't want to merge it.
-Don't take this personally - some PRs are fantastic, but just in the wrong place
-or proposed at the wrong time.
-- If your code passes code review, your changes will be approved. They may then
-be merged into the main Fusion codebase at a later time, subject to conditions
-around what we intend to include in different Fusion versions.
-	- Sometimes, we may decide to merge your code, but disable it behind a flag.
-	This is often used to disable features we don't want to make public until
-	some other pull requests are implemented or further design work is done.
-- We always make sure to credit pull requests and their contributors in release
-notes.
\ No newline at end of file
+
+### Are you the first person to suggest it?
+
+- Make sure to search for open issues *and* closed issues - it's possible
+your idea was already suggested and rejected.
+
+- If you have a substantial case for why it should not have been rejected,
+feel free to add to the existing issue with your thoughts.
+
+- If an issue is found to be a duplicate, it is closed right away. We don't
+want to split discussion about an issue across multiple locations.
+
+### Open a new issue, describing the feature request
+
+- Your title should be clear, concise, and adequately summarise the idea.
+
+- Avoid exclusively talking about specific solutions to problems - your
+feature request should describe the general case for why a feature should be
+added. Focus on who your feature request would help, when it would help them
+and why.
+
+- That said, please feel free to suggest hypothetical API designs, as long
+as they're not the focus.
+
+### We'll work with you to develop it further
+
+- Most feature requests originate from valid concerns, but might need work to
+craft an API that's suitable for Fusion.
+
+- We'll close the issue once an implementation of the design is approved and
+merged into a main branch of Fusion.
+
+### Tips for good feature requests
+
+- We typically prefer users declaring information upfront, rather than
+procedural solutions that depend on execution order.
+
+- Guide users towards writing code that's more optimal, but not at the expense
+of maintainability.
+
+- Aim to make errors impossible to represent, or at least very easy to spot.
+
+### Tips for bad feature requests
+
+- Overly specific solutions. Niche features often would be better served as
+part of third party libraries; Fusion deals with concerns that everyone has an
+interest in.
```

Comment:
The header is "tips for bad feature requests", but the points are only stated as things that _make_ for bad feature requests.